backends/imgui_impl_vulkan.cpp

```diff
@@ -1699,8 +1773,33 @@ static void ImGui_ImplVulkan_CreateWindow(ImGuiViewport* viewport)
     vd->WindowOwned = true;
 
     // Create pipeline (shared by all secondary viewports)
-    if (bd->PipelineForViewports == VK_NULL_HANDLE)
-        ImGui_ImplVulkan_CreatePipeline(v->Device, v->Allocator, VK_NULL_HANDLE, wd->RenderPass, VK_SAMPLE_COUNT_1_BIT, &bd->PipelineForViewports, 0);
+	if (bd->PipelineForViewports == VK_NULL_HANDLE)
+	{
+		ImGui_ImplVulkan_PipelineCreateInfo pci;
+		pci.Device = v->Device;
+		pci.Allocator = v->Allocator;
+		pci.PipelineCache = VK_NULL_HANDLE;
+		pci.RenderPass = wd->RenderPass;
+		pci.Subpass = 0;
+		pci.MSAASamples = VK_SAMPLE_COUNT_1_BIT;
+#ifdef IMGUI_IMPL_VULKAN_HAS_DYNAMIC_RENDERING
+		VkPipelineRenderingCreateInfoKHR rendering_info = {};
+		if (wd->UseDynamicRendering)
+		{
+			rendering_info.sType = VK_STRUCTURE_TYPE_PIPELINE_RENDERING_CREATE_INFO;
+			rendering_info.pNext = nullptr;
```

Comment:
same thing with render_info. Surely there is some cross impl pattern in the data structure that make it worth creating a factory.